src/blockdev.rs

```diff
@@ -519,6 +519,13 @@ pub struct SavedPartitions {
     partitions: Vec<(u32, GPTPartitionEntry)>,
 }
 
+fn disk_has_mbr(disk: &mut File) -> bool {
+    let mut sig = [0u8; 2];
+    disk.seek(SeekFrom::Start(510)).unwrap();
+    disk.read_exact(&mut sig).unwrap();
```

Comment:
`unwrap` is only suitable if a failure is a programmer error, or in tests.  These should use `context()?` and return `Result<bool>`.